lib/Dialect/AIEX/Transforms/AIEDMATasksToNPU.cpp

```diff
@@ -260,6 +273,40 @@ struct AIEDMATasksToNPUPass : AIEDMATasksToNPUBase<AIEDMATasksToNPUPass> {
         input_sizes[i] = (*dims)[j].getSize();
         input_strides[i] = (*dims)[j].getStride();
       }
+      if (dims->size() > 2) {
+        d2size = (target_model.isMemTile(tile.getCol(), tile.getRow()))
+                     ? (*dims)[2].getSize()
+                     : 0;
+      }
+      // if(failed(verifyPaddingDims(target_model, tile.getCol(), tile.getRow(),
+      // dims, padDims))) return failure();
+      if (padDims.has_value()) {
+        if (!target_model.isMemTile(tile.getCol(), tile.getRow()))
+          return bd_op->emitOpError()
+                 << "Padding is only supported by memtile dma bds.";
+        if (padDims->size() > dims->size())
+          return bd_op->emitOpError()
+                 << "Mismatch number of dimensions between padding(s)"
+                 << " and wrap(s) and stride(s).";
+      }
+
+      if (target_model.isMemTile(tile.getCol(), tile.getRow()) &&
```

Comment:
This if could be combined with the one above which already throws an error if we are not on a memtile.